code/game/objects/items/attachments/laser_sight.dm

```diff
@@ -21,3 +21,22 @@
 		gun.wield_delay += 0.3 SECONDS
 
 	playsound(user, toggled ? 'sound/weapons/magin.ogg' : 'sound/weapons/magout.ogg', 40, TRUE)
+
+/obj/item/attachment/laser_sight/on_preattack(obj/item/gun/gun, atom/target, mob/user, list/params)
```

Comment:
im cutting this, it didn't work how i thought it would

i thought it woudl palce a laser dot where you fired, but this only seems to proc when attacking something next to you

Reply:
No, actually. If you fire with an empty mag, it will point too.

Reply:
Or... maybe not, nevermind? Thought it did last uptime.